pkg/cmd/dedicated/register/registercluster.go

```diff
@@ -181,13 +182,19 @@ func runClusterSelectionInteractivePrompt(opts *options) error {
 	return nil
 }
 
-// parses the
+// parseDNSURL attempts to parse the cluster ingress dns name from the console url.
 func parseDNSURL(opts *options) (string, error) {
 	clusterIngressDNSName := opts.selectedCluster.Console().URL()
 	if len(clusterIngressDNSName) == 0 {
 		return "", fmt.Errorf("DNS url is empty")
 	}
-	return strings.SplitAfter(clusterIngressDNSName, ".apps.")[1], nil
+
+	splits := strings.SplitAfterN(clusterIngressDNSName, "console-openshift-console.", 2)
+	if len(splits) == 2 {
+		return splits[1], nil
+	}
+
+	return "", fmt.Errorf("could not construct cluster_ingress_dns_name")
 }
```

Comment:
The logic might not be needed from the CLI in the next releases of KFM - i.e KFM might take over this logic and do it on a case by case basis.